### PR TITLE
Fix py3 strings

### PR DESCRIPTION
**Pull request by newville**
this should fix several issues with string/byte values in string and waveform PVs and Python3, including #22. 

I have not yet run a full test suite (hard for me tor do during active data collection and real operations), so will wait to merge.

### CODE REVIEW

**Comment by tacaswell** [comment]
Great, we are in a studies period, hopefully I will be able to get to exercise this next week.

**Comment by tacaswell** [comment]
@newville This needs a rebase

File: make_wininsts.bat [hunk 1]
```diff
@@ -1,10 +1,10 @@
-SET PATH=C:\Python26;%PATH%
```
Comment (tacaswell) [inline make_wininsts.bat]: What happened to this file, the rest of the `*.bat`, and `st.cmd` ?  I assume there is line ending change, was that intentional?

Comment (newville) [inline make_wininsts.bat]: sorry, I'm not sure I understand the question.  Are the line endings not auto-converted for you?

Comment (tacaswell) [inline make_wininsts.bat]: They are showing up as every line change in these diffs with no obvious change which tends to mean line endings.  It's not really a problem, just super thrashy / messes with git-blame.

Comment (klauer) [inline make_wininsts.bat]: Add ?w=1 to the diff url to remove whitespace-only changes. Compare:
https://github.com/pyepics/pyepics/pull/35/files?w=1
with
https://github.com/pyepics/pyepics/pull/35/files

Comment (tacaswell) [inline make_wininsts.bat]: @klauer I am not concerned about seeing the thrashing, I am concerned that it is happening.

Comment (klauer) [inline make_wininsts.bat]: @tacaswell Agreed, though my comment was mainly for giving others context.

Comment (tacaswell) [inline make_wininsts.bat]: fair, sorry for being snippy.

Comment (newville) [inline make_wininsts.bat]: would setting '* text=auto" in .gitattributes be preferred?  I have to
admit I've never really been that picky about line endings, or maybe just
used to working in a mixed Win/Unix environment.

On Thu, Oct 1, 2015 at 10:31 AM, Thomas A Caswell <notifications@github.com>
wrote:

> In make_wininsts.bat
> <https://github.com/pyepics/pyepics/pull/35#discussion_r40927207>:
>
> > @@ -1,10 +1,10 @@
> > -SET PATH=C:\Python26;%PATH%
>
> fair, sorry for being snippy.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyepics/pyepics/pull/35/files#r40927207>.
>



-- 
--Matt Newville <newville at cars.uchicago.edu> 630-252-0431

**Pull request closed without merging**